cyral/internal/policyv2/datasource.go

```diff
@@ -0,0 +1,112 @@
+
+import (
+	"fmt"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+
+	"github.com/cyralinc/terraform-provider-cyral/cyral/client"
+	"github.com/cyralinc/terraform-provider-cyral/cyral/core"
+	"github.com/cyralinc/terraform-provider-cyral/cyral/core/types/resourcetype"
+)
+
+var dsContextHandler = core.DefaultContextHandler{
+	ResourceName:                 dataSourceName,
+	ResourceType:                 resourcetype.DataSource,
+	SchemaWriterFactoryGetMethod: func(_ *schema.ResourceData) core.SchemaWriter { return &PolicyV2{} },
+	ReadUpdateDeleteURLFactory: func(d *schema.ResourceData, c *client.Client) string {
+		return fmt.Sprintf("https://%s/%s/%s", c.ControlPlane, getAPIPath(d.Get("type").(string)), d.Get("id").(string))
+	},
+}
+
+func dataSourceSchema() *schema.Resource {
+	return &schema.Resource{
+		Description: "Some description.",
+		ReadContext: dsContextHandler.ReadContext(),
+		Schema: map[string]*schema.Schema{
+			"id": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"name": {
+				Description: "Retrieve the unique label with this name, if it exists.",
+				Type:        schema.TypeString,
+				Optional:    true,
+			},
+			"description": {
+				Description: "Description of the data source.",
+				Type:        schema.TypeString,
+				Optional:    true,
+			},
+			"enabled": {
+				Description: "Indicates if the policy is enabled.",
+				Type:        schema.TypeBool,
+				Optional:    true,
+			},
+			"scope": {
+				Description: "Repos that the policy applies to, empty value implies applicability to all repos.",
+				Type:        schema.TypeList,
+				Optional:    true,
+				Elem:        &schema.Schema{Type: schema.TypeString},
+			},
+			"tags": {
+				Description: "Tags associated with the policy to categorize it.",
+				Type:        schema.TypeList,
+				Optional:    true,
+				Elem:        &schema.Schema{Type: schema.TypeString},
+			},
+			"valid_from": {
+				Description: "Time when the policy comes into effect.",
+				Type:        schema.TypeString,
+				Optional:    true,
+			},
+			"valid_until": {
+				Description: "Time after which the policy is no longer in effect.",
+				Type:        schema.TypeString,
+				Optional:    true,
+			},
+			"document": {
+				Description: "The actual policy document in JSON format.",
+				Type:        schema.TypeString,
+				Optional:    true,
+			},
+			"last_updated": {
+				Description: "Information about when and by whom the policy was last updated.",
+				Type:        schema.TypeMap,
+				Optional:    true,
+				Elem: &schema.Schema{
+					Type: schema.TypeString,
+				},
+			},
+			"created": {
+				Description: "Information about when and by whom the policy was created.",
+				Type:        schema.TypeMap,
+				Optional:    true,
+				Elem: &schema.Schema{
+					Type: schema.TypeString,
+				},
+			},
+			"enforced": {
+				Description: "Indicates if the policy is enforced.",
+				Type:        schema.TypeBool,
+				Optional:    true,
```

Comment:
Is there a way to specify what the default is? Not needed if tf assumes the default is the go default for the type.

Reply:
You can specify, but looks right to me.
TF:
```
resource "cyral_policy_v2" "local_policy_example" {
  name        = "minimal_local_policy"
  document    = jsonencode({
    governedData = {
      locations = ["gym_db.users"]
    }
    readRules = [
      {
        conditions = [
          {
            attribute = "identity.userGroups"
            operator  = "contains"
            value     = "users"
          }
        ]
        constraints = {
          datasetRewrite = "SELECT * FROM $${dataset} WHERE email = '$${identity.endUserEmail}'"
        }
      }
    ]
  })
  type        = "local"
}
```

Created policy (GET from API):
```
{
    "policy": {
        "id": "5250617c-c5de-43cc-954a-15b6682e97f6",
        "name": "minimal_local_policy",
        "description": "",
        "enabled": false,
        "scope": null,
        "tags": [],
        "validFrom": null,
        "validUntil": null,
        "document": "{\"governedData\":{\"locations\":[\"gym_db.users\"]},\"readRules\":[{\"conditions\":[{\"attribute\":\"identity.userGroups\",\"operator\":\"contains\",\"value\":\"users\"}],\"constraints\":{\"datasetRewrite\":\"SELECT * FROM ${dataset} WHERE email = '${identity.endUserEmail}'\"}}]}",
        "lastUpdated": null,
        "created": {
            "actor": "sa/default/de84bc5b-8204-43cf-a6ab-e101db2c67f4",
            "actorType": "ACTOR_TYPE_API_CLIENT",
            "timestamp": "2024-06-12T12:53:44.310243758Z"
        },
        "enforced": false
    }
}
```

Reply:
There is a `Default` attribute in the `Schema` that can be used for that.